servicetalk-dns-discovery-netty/src/main/java/io/servicetalk/dns/discovery/netty/DefaultDnsServiceDiscovererBuilder.java

```diff
@@ -170,11 +212,10 @@ public DefaultDnsServiceDiscovererBuilder ttl(final int minSeconds, final int ma
                     " (expected: 0 <= minCacheSeconds <= minSeconds(" + minSeconds +
                     ") <= maxCacheSeconds <= maxSeconds(" + maxSeconds + "))");
         }
-        this.minTTLSeconds = minSeconds;
-        this.maxTTLSeconds = maxSeconds;
-        this.minTTLCacheSeconds = minCacheSeconds;
-        this.maxTTLCacheSeconds = maxCacheSeconds;
-        return this;
+        if (negativeTTLCacheSeconds < 0) {
```

Comment:
this is called in a static context and will prevent the app from starting which maybe difficult to understand. did you consider logging an error and choosing a default value if invalid values are specified?

Reply:
Actually, it's a leftover from some intermediate work when I had more properties. With only one property, there is no need to validate inside the static block as we always choose the default, even if we can not parse the value.